expression/constant_propagation.go

```diff
@@ -317,6 +318,15 @@ func (s *propConstSolver) pickNewEQConds(visited []bool) (retMapper map[int]*Con
 		if col.GetType().Hybrid() {
 			continue
 		}
+
+		// The cond must be eq ScalarFunction if the program run here.
+		// Only the deterministic expression can propagation. e.g. [a = 'A'] if it use non-binary collation to compare, a could be 'a' or 'A', or 'a  ',
```

Comment:
Will this cause regression?
If there's `a = b and a = 'a'` where a and b are both general_ci, we can infer `b='a'` though a is not bin collation?

Reply:
it will cause regression in this case, so we do not merge this PR until now